src/transformers/models/speech_encoder_decoder/modeling_speech_encoder_decoder.py

```diff
@@ -223,7 +223,9 @@ def __init__(
         self.encoder.config = self.config.encoder
         self.decoder.config = self.config.decoder
 
-        if self.encoder.config.hidden_size != self.decoder.config.hidden_size:
+        # get encoder output hidden size
+        self.encoder_output_dim = getattr(config.encoder, "output_hidden_size", config.encoder.hidden_size)
```

Comment:
We need to stretch this rule a bit here...the hidden size does not always correspond to the encoder output hidden size and it's the encoder output hidden size that matters

Reply:
LGTM